pkg/resources/usercluster/deployment.go

```diff
@@ -171,7 +171,7 @@ func DeploymentCreator(data userclusterControllerData) reconciling.NamedDeployme
 			}
 
 			if data.Cluster().Spec.OPAIntegration != nil && data.Cluster().Spec.OPAIntegration.ExperimentalEnableMutation {
-				args = append(args, "--enable-mutation", fmt.Sprint(&data.Cluster().Spec.OPAIntegration.ExperimentalEnableMutation))
+				args = append(args, "-enable-mutation", fmt.Sprint(data.Cluster().Spec.OPAIntegration.ExperimentalEnableMutation))
```

Comment:
https://pkg.go.dev/flag#hdr-Command_line_flag_syntax

```
-flag x  // non-boolean flags only
```

please change to 
```
fmt.Sprintf("-enable-mutation=%t", data.Cluster().Spec.OPAIntegration.ExperimentalEnableMutation)),
```

Reply:
btw. should not be the condition the same as here [link](https://github.com/kubermatic/kubermatic/pull/7749/files#diff-fc8f73fda439bfb46949c7196ce280cfe5393c27b629249b8d342a5ff4f981f3R173)?

If OPAIntegration is disabled but experimental mutation is enabled, do we still want to run the mutation webhook? If yes, why?

Reply:
In that case we don't want mutation webhook, updated the changes.
fyi create a PR to remove mutation webhook when `enable-mutation:false`